src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -99,18 +99,55 @@
 
     @Override
     public Connection connect(String url, Properties info) throws SQLException {
+        Pair<MongoConnection, Integer> p = getUnvalidatedConnectionAndTimeout(url, info);
+        Connection conn = p.left();
```

Comment:
```
        Pair<MongoConnection, Integer> p = getUnvalidatedConnectionAndTimeout(url, info);
        if (p == null) {
            return null;
         }
        Connection conn = p.left();
```
`p` is returned as null for a wrong URL , p.left() would still throw NPE

Reply:
Wow, I did that too fast.

Reply:
(I'm changing it to return `new Pair<>(null, 0)`)

Reply:
Actually, after looking at @nbagnard's suggestion, I was able to clean thus up even more so we don't have to deal with `Pair` at all :)